tools/download_coder_imatrix_datasets.py

```
@@ -0,0 +1,217 @@
+    code_lines = []
+
+    # The Stack v2 - top languages only (Python, JS, TS, Java, C++, Go, Rust, C#, PHP, Ruby)
+    stack_langs = ["Python", "JavaScript", "TypeScript", "Java", "C++", "Go", "Rust", "C#", "PHP", "Ruby"]
+    for lang in stack_langs:
+        lines = read_or_generate(
+            "bigcode/the-stack-v2-dedup",
+            split="train",
+            text_key="content",
+            max_samples=3000,  # ~30k total
+            min_length=30,
+            filter_fn=lambda x: x.get("lang") == lang and x.get("size") > 100
```

Ruff (0.14.14)

[warning] Function definition does not bind loop variable `lang`

(B023)